Adding commit comment functionality

ipv6freely opened the pull request:
Adding a commit comment to the junos driver is incredibly simply since the PyEZ commit() function takes a comment natively.

I noticed that in some other branches it seems napalm devs want to use "message" instead of "comment", so that's what I used. Have tested this locally already and works perfectly.

I also added the message=None argument to the commit_config() function for the other drivers to. Most don't do anything, but IOS-XR will raise an exception when the message is used because it hasn't yet been full implemented.

dbarrosop commented on napalm/iosxr/iosxr.py:
```diff
@@ -145,7 +145,9 @@ def compare_config(self):
-    def commit_config(self):
+    def commit_config(self, message=None):
+        if message is not None:
+            raise ValueError('Commit message not yet implemented for IOS-XR')
```
Could you change this to `NotImplementedException`?

dbarrosop replied:
Would you also mind adding this same check to all the drivers?

dbarrosop commented on napalm/iosxr/iosxr.py:
```diff
@@ -145,7 +145,9 @@ def compare_config(self):
             return self.device.compare_config().strip()
 
-    def commit_config(self):
+    def commit_config(self, message=None):
```
I think `message=""` as it also gives away the expected type.

ktbyers replied:
@dbarrosop I don't have any objections to this, if this changed; then the exception would also need to be changed to:

```
if not message:
    raise NotImplementedException
```

ipv6freely replied:
You would prefer `message=""` instead of `message=None` ?

dbarrosop commented:
@mirceaulinic @ktbyers I think it's fine we proceed with this PR with the minor changes I requested. What do you think?

ktbyers commented:
Yes, I think we should move forward with the changes you proposed. As this will give @ipv6freely what he needs to use this.

ipv6freely commented:
Sounds good, I'll make those changes.

ipv6freely commented:
I made the requested changes. I added `message=''` to all of the methods (using single quotes as that seems to be what is being used elsewhere) and also changed to `raise NotImplementedError` with a generic message of `Commit message not implemented for this platform`

I believe we're good to go at this point.

mirceaulinic commented on napalm/iosxr/iosxr.py:
```diff
@@ -145,7 +145,9 @@ def compare_config(self):
-    def commit_config(self):
+    def commit_config(self, message=''):
+        if not message:
+            raise NotImplementedError('Commit message not implemented for this platform')
```
Why not? It is supported by IOSXR and the underlying library: https://github.com/fooelisa/pyiosxr/blob/master/pyIOSXR/iosxr.py#L577

All you have to do is:

```python
         if self.replace:
             self.device.commit_replace_config(comment=message)
         else:
             self.device.commit_config(comment=message)
````

ipv6freely replied:
Because @dbarrosop and @ktbyers said to do it this way until somebody who knows IOS-XR can add that functionality.

ipv6freely replied:
I just added it.

mirceaulinic commented on napalm/nxos/nxos.py:
```diff
@@ -44,7 +44,9 @@
-    def commit_config(self):
+    def commit_config(self, message=''):
+        if not message:
+            raise NotImplementedError('Commit message not implemented for this platform')
```
I think we should do the same for `nxos_ssh`? @ktbyers

ipv6freely replied:
There is no `commit_config()` in `nxos_ssh`.

ipv6freely commented:
It just occured to me that the logic is backwards. it should be `if message raise NotImplementedError` because the exception should be raised if the user _has_ specified a message

ipv6freely commented:
I fixed the logic fail using `if message:`, and added support for IOS-XR per @mirceaulinic 's request. 

I also changed the methods to have `message=None` as this seems to be the way it's done everywhere else and I think it makes it more consistent.

ipv6freely commented:
@mirceaulinic Do you have everything you need for this, now?

ktbyers commented:
I have some changes I am going to make on this PR. I will update it.

ktbyers commented:
I updated this PR here: https://github.com/napalm-automation/napalm/pull/701
